### PR TITLE
Rewrited service.py

### PR DESCRIPTION
**Pull request by antl31**
Upgraded, fixed errors

### CODE REVIEW

File: tests/tests.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+try:
```
Comment (iivanou) [inline tests/tests.py]: I'd suggest renaming this module to test_service.py. Don't think you are planning to put all the tests into one file... I hope so:)

File: tests/tests.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+from reportportal_client.service import _convert_string, _list_to_payload, \
+    uri_join, _get_id, _get_msg, _get_data, _get_json, _get_messages, ReportPortalService
+
+import unittest
+from datetime import datetime
+
+
+class Test(unittest.TestCase):
+
+    def test_check_convert_to_string(self):
```
Comment (iivanou) [inline tests/tests.py]: Let's start adding docstrings everywhere. See PEP 257.

File: tests/tests.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        self.rp.session = MagicMock()
+
+    def test_start_launch(self):
+        with patch('reportportal_client.service._get_data', new_callable=Mock()) as mock_get:
+            mock_get.return_value = {"id": 111}
+            launch_id = self.rp.start_launch('name', datetime.now().isoformat())
+        self.assertEqual(launch_id, 111)
+
+    def test_finish_launch(self):
+        with patch('reportportal_client.service._get_msg', new_callable=MagicMock()) as mock_get:
```
Comment (iivanou) [inline tests/tests.py]: Why do u use the MagicMock here and the regular Mock at line 82? It seems that the simple Mock would be enough.

File: tests/tests.py (hunk 2 above)
Comment (antl31) [reply tests/tests.py]: tests usuallly are not describe, what you want to see in docstrings?

File: reportportal_client/service.py [hunk 4]
```diff
@@ -28,15 +28,16 @@
 logger.addHandler(logging.NullHandler())
 
 
-def _dict_to_payload(dictionary):
-    def _str(value):
-        if isinstance(value, six.text_type):
-            # Don't try to encode 'unicode' in Python 2.
-            return value
-        return str(value)
+def _convert_string(value):
```
Comment (iivanou) [inline reportportal_client/service.py]: Docstring is missing.

File: reportportal_client/service.py [hunk 5]
```diff
@@ -28,15 +28,16 @@
...
+    if isinstance(value, six.text_type):
+        # Don't try to encode 'unicode' in Python 2.
+        return value
+    return str(value)
 
+
+def _list_to_payload(dictionary):
     return [
```
Comment (iivanou) [inline reportportal_client/service.py]: Docstring is missing.

File: tests/test_service.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
+try:
```
Comment (iivanou) [inline tests/test_service.py]: Module docstring is missing.

File: tests/test_service.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        :return: (int) id
+        """
+        with patch('reportportal_client.service._get_data',
+                   new_callable=Mock()) as mock_get:
+            mock_get.return_value = {"id": 111}
+            launch_id = self.rp.start_launch('name',
+                                             datetime.now().isoformat())
+        self.assertEqual(launch_id, 111)
+
+    """Testing for finish launch and sending request"""
```
Comment (iivanou) [inline tests/test_service.py]: I think it should be a part of the docstring of the test_finish_launch method.

File: tests/test_service.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+        self.assertEqual(obj, ['422: error'])
+
+
+class ReportPortalServiceTest(unittest.TestCase):
+    def setUp(self):
+        """Creating ReportPortalService Instance and Mocking session"""
+        self.rp = ReportPortalService('http://endpoint', 'project', 'token')
+        self.rp.session = MagicMock()
+
+    """Testing for start launch and sending request"""
```
Comment (iivanou) [inline tests/test_service.py]: I think it should be a part of the docstring of the test_statr_launch method.

File: tests/test_service.py [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    uri_join, _get_id, _get_msg, _get_data, _get_json, _get_messages, \
+    ReportPortalService
+
+import unittest
+from datetime import datetime
+
+
+class TestServiceFunctions(unittest.TestCase):
+    """
+    Test for main functions in service.py
```
Comment (iivanou) [inline tests/test_service.py]: One-line docstring should fit on one line with quotes.  https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

File: tests/test_service.py [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+            mock_get.status_code = 200
+            mock_get.json.return_value = fake_json
+
+            obj = _get_json(mock_get)
+
+        self.assertEqual(obj, fake_json)
+
+    def test_get_messages(self):
+        """Test for get message from response
+        :param: Response
```
Comment (iivanou) [inline tests/test_service.py]: 1 blank line required between summary line and description. First line should end with a period. https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

File: tests/test_service.py [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+        :return: list
+        """
+        data = {"responses": [{"errorCode": 422, "message": "error"}]}
+
+        obj = _get_messages(data)
+
+        self.assertEqual(obj, ['422: error'])
+
+
+class ReportPortalServiceTest(unittest.TestCase):
```
Comment (iivanou) [inline tests/test_service.py]: Docstring is missing.

File: reportportal_client/service.py [hunk 12]
```diff
@@ -244,6 +253,17 @@ def finish_test_item(self,
                          issue=None,
                          attributes=None,
                          **kwargs):
+        """
```
Comment (iivanou) [inline reportportal_client/service.py]:
```suggestion
        """ Finish the test item and return HTTP response.
       
        :param item_id:    id of the test item
        :param end_time:   time in UTC format
        :param status:     status of the test
        :param issue:      description of an issue
        :param attributes: list of attributes
        :param kwargs:     other parameters
        :return:           json message
        """
```

Please update similar doc strings the same way.

File: tests/test_service.py [hunk 13]
```diff
@@ -12,34 +14,25 @@
...
-        """Test for get id from Response obj"""
+        """Test for get id from Response obj."""
         fake_json = {"id": 123}
 
-        """
-        Mocking Response obj
-        :param status_code 200
-        :param return JSON obj fake_json
-        """
+        """Mocking Response obj."""
```
Comment (iivanou) [inline tests/test_service.py]:
```suggestion
       # Mocking the Response object.
```

File: tests/test_service.py [hunk 14]
```diff
@@ -118,17 +98,21 @@ def test_get_messages(self):
 
 
 class ReportPortalServiceTest(unittest.TestCase):
+    """Tests for ReportPortalService."""
+
     def setUp(self):
-        """Creating ReportPortalService Instance and Mocking session"""
+        """Create ReportPortalService Instance and Mocking session."""
```
Comment (iivanou) [inline tests/test_service.py]:
```suggestion
        """Instantiate the ReportPortalService class and mock its session."""
```

File: reportportal_client/service.py (hunk 4 above)
Comment (iivanou) [reply reportportal_client/service.py]: Still missing.

File: reportportal_client/service.py (hunk 5 above)
Comment (iivanou) [reply reportportal_client/service.py]: Still missing.

File: reportportal_client/service.py [hunk 15]
```diff
@@ -152,6 +155,7 @@ def __init__(self,
         self.verify_ssl = verify_ssl
 
     def terminate(self, *args, **kwargs):
+        """terminate."""
```
[the remark this answers is not in the document]
Comment (iivanou) [reply reportportal_client/service.py]: Still actual...

File: reportportal_client/service.py [hunk 16]
```diff
@@ -178,7 +183,9 @@ def start_launch(self,
 
     def finish_launch(self, end_time, status=None, **kwargs):
         """
```
[the remark this answers is not in the document]
Comment (iivanou) [reply reportportal_client/service.py]: Still actual...

File: reportportal_client/service.py (hunk 12 above)
Comment (iivanou) [reply reportportal_client/service.py]: Still actual...

File: tests/test_service.py [hunk 17]
```diff
@@ -0,0 +1,126 @@
+"""Tests for service.py ."""
```
Comment (iivanou) [inline tests/test_service.py]: This module contains unit tests for the service.py module.

File: tests/test_service.py [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+
+            obj = _get_id(mock_get)
+
+        self.assertEqual(obj, 123)
+
+    def test_get_msg(self):
+        """
+        Test get_msg recieved from Response.
+
+        :return dict
```
Comment (iivanou) [inline tests/test_service.py]: Your test function does not return anything(Well, actually it returns None). 

"""Test get_msg recieved from Response."""

File: tests/test_service.py [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+        with patch('requests.Response', new_callable=MagicMock()) as mock_get:
+            mock_get.status_code = 200
+            mock_get.json.return_value = fake_json
+
+            obj = _get_msg(mock_get)
+
+        self.assertEqual(obj, fake_json)
+
+    def test_get_data(self):
+        """
```
Comment (iivanou) [inline tests/test_service.py]: Your test function does not return anything(Well, actually it returns None).

File: tests/test_service.py [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+        with patch('requests.Response', new_callable=MagicMock()) as mock_get:
+            mock_get.status_code = 200
+            mock_get.json.return_value = fake_json
+
+            obj = _get_data(mock_get)
+
+        self.assertEqual(obj, fake_json)
+
+    def test_get_json(self):
+        """Test get json from Response.
```
Comment (iivanou) [inline tests/test_service.py]: Your test function does not return anything(Well, actually it returns None).

File: tests/test_service.py [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+            mock_get.json.return_value = fake_json
+
+            obj = _get_json(mock_get)
+
+        self.assertEqual(obj, fake_json)
+
+    def test_get_messages(self):
+        """Test get errors from response.
+
+        :return: list
```
Comment (iivanou) [inline tests/test_service.py]: Your test function does not return anything(Well, actually it returns None).

File: tests/test_service.py [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+        """
+        data = {"responses": [{"errorCode": 422, "message": "error"}]}
+
+        obj = _get_messages(data)
+
+        self.assertEqual(obj, ['422: error'])
+
+
+class ReportPortalServiceTest(unittest.TestCase):
+    """Tests ReportPortalService."""
```
Comment (iivanou) [inline tests/test_service.py]: """This class stores methods which test ReportPortalService."""

File: tests/test_service.py (hunk 14 above)
Comment (iivanou) [reply tests/test_service.py]: Still actual.

File: .travis.yml [hunk 23]
```diff
@@ -13,6 +13,7 @@ install:
 before_script:
   - pycodestyle .
 script:
+  - python setup.py test
```
Comment (iivanou) [inline .travis.yml]: install:
  - pip install pycodestyle pydocstyle
before_script:
  - pycodestyle .
  - pydocstyle .

**Pull request merged**